inference-engine/tests/functional/shared_test_classes/include/shared_test_classes/single_layer/experimental_detectron_generate_proposals_single_image.hpp

```diff
@@ -0,0 +1,47 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "ngraph_functions/utils/ngraph_helpers.hpp"
+#include "common_test_utils/common_utils.hpp"
+#include "shared_test_classes/base/ov_subgraph.hpp"
+
+namespace ov {
+namespace test {
+namespace subgraph {
+
+typedef std::tuple<
+        std::vector<InputShape>,                // Input shapes
+        float,                                  // min_size: minimum box width & height
+        float,                                  // nms_threshold: specifies NMS threshold
+        int64_t,                                // post_nms_count: number of top-n proposals after NMS
+        int64_t,                                // pre_nms_count: number of top-n proposals after NMS
+        std::pair<std::string, std::vector<ov::runtime::Tensor>>, // input tensors
+        ElementType,                            // Network precision
+        std::string                             // Device name>;
+> ExperimentalDetectronGenerateProposalsSingleImageTestParams;
+
+class ExperimentalDetectronGenerateProposalsSingleImageLayerTest :
+        public testing::WithParamInterface<ExperimentalDetectronGenerateProposalsSingleImageTestParams>,
+        virtual public SubgraphBaseTest {
+protected:
+    void SetUp() override;
+    void generate_inputs(const std::vector<ngraph::Shape>& targetInputStaticShapes) override;
+
+public:
+    static std::string getTestCaseName(const testing::TestParamInfo<ExperimentalDetectronGenerateProposalsSingleImageTestParams>& obj);
+    template <class T>
+    static ov::runtime::Tensor createTensor(
```

Comment:
Looks like should be moved to `data_utils`

Reply:
can I leave it here and don't update test utils?

Reply:
done, refactoring was done. Thanks!